golang/cosmos/x/swingset/types/params.go

```diff
@@ -163,3 +163,73 @@ func validateQueueMax(i interface{}) error {
 	}
 	return nil
 }
+
+// UpdateParams adds any new params that are missing, configuring them to their
+// defaults, then returning the updated params or an error.
+func UpdateParams(params Params) (Params, error) {
+	newBpu, err := addDefaultBeansPerUnit(params.BeansPerUnit, DefaultBeansPerUnit())
+	if err != nil {
+		return params, err
+	}
+	newPff, err := addDefaultPowerFlagFees(params.PowerFlagFees, DefaultPowerFlagFees)
+	if err != nil {
+		return params, err
+	}
+	newQm, err := addDefaultQueueSize(params.QueueMax, DefaultQueueMax)
+	if err != nil {
+		return params, err
+	}
+
+	params.BeansPerUnit = newBpu
+	params.PowerFlagFees = newPff
+	params.QueueMax = newQm
+	return params, nil
+}
+
+// addDefaultBeansPerUnit adds the default beans per unit entries not in the
+// list of bean costs already, returning the possibly-updated list, or an error.
+func addDefaultBeansPerUnit(bpu []StringBeans, defaultBpu []StringBeans) ([]StringBeans, error) {
```

Comment:
It took me some studying to figure out how this function actually did the work.  Maybe consider the following changes to this function's name, and similar for `appendMissingDefaultPowerFlagFee` and `appendMissingDefaultQueueSize`:
```suggestion
// appendMissingDefaultBeansPerUnit appends the default beans per unit entries not in the
// list of bean costs already, returning the possibly-updated list, or an error.
func appendMissingDefaultBeansPerUnit(bpu []StringBeans, defaultBpu []StringBeans) ([]StringBeans, error) {
```